vcd/resource_vcd_independent_disk.go

```diff
@@ -128,7 +199,18 @@ func resourceVcdIndependentDiskCreate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("error waiting to finish creation of independent disk: %s", err)
 	}
 
-	d.SetId(diskName)
+	diskHref := task.Task.Owner.HREF
+	disk, err := vdc.GetDiskByHref(diskHref)
+	if govcd.IsNotFound(err) {
+		log.Printf("unable to find disk with href %s: %s. Removing from state", diskHref, err)
+		d.SetId("")
+		return nil
```

Comment:
In the creation method, failure to find the disk for any reason should result in an explicit error, not a state removal

Reply:
updated